### ISSUE TITLE
"Open File" when in replace preview diff mode does not work

### ISSUE THREAD

**Issue by komputerwiz** [Microsoft/vscode#35323]
- VSCode Version: Code 1.16.1 (27492b6bf3acb0775d82d2f87b25a93490673c6d, 2017-09-14T16:26:20.018Z)
- OS Version: Linux x64 4.4.0-96-generic (also affects macOS 10.2.6 and Windows 10 Enterprise)
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
EditorConfig|Edi|0.10.1
gc-excelviewer|Gra|1.1.15
r|Iku|0.4.6
python|don|0.7.0
git-project-manager|fel|1.4.0
php-debug|fel|1.11.1
php-intellisense|fel|1.5.1
phpcs|ika|0.7.0
Go|luk|0.6.65
ruby-rubocop|mis|0.4.0
cpptools|ms-|0.13.0
php-docblocker|nei|1.2.0
vscode-subword-navigation|ow|1.1.5
Ruby|reb|0.15.0
java|red|0.10.0
vscode-icons|rob|7.14.0
twig|wha|1.0.2
change-case|wma|1.0.0

---

Steps to Reproduce:

1. Open "Search" pane and enter search/replace values
2. Click result to open "Replace Preview" editor
3. Click "Open File" at right of tab bar to open file for editing

Expected Result: An editable tab opens to the selected file.

Actual Result: No response.

Reproduces without extensions: Yes

**Comment by komputerwiz** [comment]
After conducting a little more research, the "Open File" contribution is provided by the git extension, and its visibility is controlled by the following logic in `extensions/git/package.json` (on branch `master` as of commit 8b1909de91f83ac305d2e113762658a99482dd10):

```json
{
  "contributes": {
    "menus": {
      "editor/title": [
        {
          "command": "git.openFile",
          "group": "navigation",
          "when": "config.git.enabled && gitOpenRepositoryCount != 0 && isInDiffEditor && resourceScheme != extension && resourceScheme != merge-conflict.conflict-diff"
        }
      ]
    }
  }
}
```

Hence this control does not apply if the project is not a git repository. The _cause of the issue_ can be addressed by updating the logic above to hide this control in the Replace Preview editor.

However, the _fundamental UX issue_ remains: suppose that, while performing a project-wide find/replace, the user would like to edit something else in the file. In my case I needed to add an import statement required by the replacement value. Since the editor(s) in the Replace Preview diff are not editable, the only way I found to open the file for editing while in "find/replace mode" was to manually do so from the explorer sidebar or quick-open palette.

After some experimenting, I noticed that the file remains opened for editing at the completion of performing a find/replace on that file, but it would be very nice to have a means of editing a given file as part of the find/replace workflow.

Would you like me to submit a separate feature request to address the UX issue above?

By the way, thanks for making such a great code editor! :smile:

**Comment by roblourens** [comment]
Yeah I agree that there are two issues:

- Git shouldn't contribute the Open File button to the replace diff, cc @joaomoreno 
- There should be a working Open File button that always appears, git repo or not, in the replace diff editor

**Comment by komputerwiz** [comment]
Roger. I've opened a separate feature request (referenced above).

**Comment by joaomoreno** [comment]
> Git shouldn't contribute the Open File button to the replace diff

I don't think so. The real problem here is that we have a single `isInDiffEditor` context. There's no clue as to whether this is a diff editor coming from git, comparing two files or the replace editor. We need to enrich the context, so the git contributed action only shows up when it's a _git_ diff editor (meaning: the left-hand side is a git buffer).
